plugins/modules/idrac_license.py

```diff
@@ -0,0 +1,577 @@
+    def __init__(self, idrac, module):
+        self.idrac = idrac
+        self.module = module
+
+    def execute(self, module):
+        license_id = module.params.get('license_id')
+        delete_license_url = f"/redfish/v1/LicenseService/Licenses/{license_id}"
+        delete_license_status = self.idrac.invoke_request(delete_license_url, 'DELETE')
+        status = delete_license_status.status_code
+        share_type = module.params.get('share_parameters').get('share_type')
+        if status == 204:
+            module.exit_json(msg=SUCCESS_DELETE_MSG, changed=True)
+        else:
+            module.fail_json(FAILURE_MSG.format(operation=share_type, license_id=license_id), changed=False)
+        return delete_license_status
+
+
+class ExportLicense():
+    def __init__(self, idrac, module):
+        self.idrac = idrac
+        self.module = module
+
+    def execute(self, module):
+        share_type = module.params.get('share_parameters').get('share_type')
+        license_id = module.params.get('license_id')
+        if share_type == "local":
+            export_license_status = export_license_local(self, module)
+        elif share_type in ["http", "https"]:
+            export_license_status = export_license_http(self, module)
+            job_status = get_job_status(self, export_license_status)
+        elif share_type == "cifs":
+            export_license_status = export_license_cifs(self, module)
+            job_status = get_job_status(self, export_license_status)
+        elif share_type == "nfs":
+            export_license_status = export_license_nfs(self, module)
+            job_status = get_job_status(self, export_license_status)
+        status = export_license_status.status_code
+        if share_type in ["http", "https", "cifs", "nfs"]:
+            if status in [200, 202]:
+                module.exit_json(msg=SUCCESS_EXPORT_MSG, changed=True, job_details=job_status)
+            else:
+                module.fail_json(msg=FAILURE_MSG.format(operation=share_type, license_id=license_id), changed=False, job_details=job_status)
+        else:
+            if status in [200, 202]:
+                module.exit_json(msg=SUCCESS_EXPORT_MSG, changed=True)
+            else:
+                module.fail_json(msg=FAILURE_MSG.format(operation=share_type, license_id=license_id), changed=False)
+        return export_license_status
+
+
+def get_job_status(self, export_license_status):
+    res_uri = validate_and_get_first_resource_id_uri(
+        self.module, self.idrac, SYSTEMS_URI)
+    job_dict = {}
+    job_tracking_uri = export_license_status.headers.get("Location")
+    job_id = job_tracking_uri.split("/")[-1]
+    job_uri = IDRAC_JOB_URI.format(job_id=job_id, res_uri=res_uri[0])
+    job_resp = self.idrac.invoke_request(job_uri, 'GET')
+    job_dict = job_resp.json_data
+    job_dict = remove_key(job_dict, regex_pattern='(.*?)@odata')
+    return job_dict
+
+
+def export_license_local(self, module):
```

Comment:
move this inside export license class. any reason why kept outside of class?

Reply:
Done